src/daemon/udp_connection/dlt_daemon_udp_socket.c

```diff
@@ -0,0 +1,262 @@
+static DltDaemonClientSockInfo g_udpmulticast_addr;
+
+// **************************************************************************
+// Function   : dlt_daemon_udp_init_clientstruct
+// In Param   : UDP client_info struct to be initilzed
+// Out Param  : NIL
+// Description: client struct to be initilized to copy control/connect/disconnect
+//                client addr
+// **************************************************************************
+void dlt_daemon_udp_init_clientstruct(DltDaemonClientSockInfo *clientinfo_struct)
+{
+    if(clientinfo_struct != NULL){
+        memset(&clientinfo_struct->clientaddr, 0x00, sizeof(clientinfo_struct->clientaddr));
+        clientinfo_struct->clientaddr_size = sizeof(clientinfo_struct->clientaddr);
+        clientinfo_struct->isvalidflag = ADDRESS_INVALID; // info is invalid
+        dlt_log(LOG_INFO, "dlt_daemon_udp_init_clientstruct: client addr struct init success \n");
+    }
+    else{
+        dlt_log(LOG_ERR, "dlt_daemon_udp_init_clientstruct: NULL arg\n");
+    }
+}
+
+
+// **************************************************************************
+// Function   : dlt_daemon_udp_setmulticast_addr
+// In Param   : NIL
+// Out Param  : NIL
+// Description: set the multicast addr to global variables
+// **************************************************************************
+void dlt_daemon_udp_setmulticast_addr(DltDaemonLocal* daemon_local)
+{
+    dlt_daemon_udp_init_clientstruct(&g_udpmulticast_addr);
+//#ifdef DLT_USE_IPv6 // IPv6 Support
+/*    struct sockaddr_in6 clientaddr6;
+    clientaddr6.sin6_family = AF_INET6;
+    inet_pton(AF_INET, daemon_local->UDPMulticastIPAddress, &clientaddr6.sin6_addr.s6_addr);
+    clientaddr6.sin6_port=htons(daemon_local->UDPMulticastIPPort);
+    memcpy(&g_udpmulticast_addr.clientaddr, &clientaddr6, sizeof(struct sockaddr_in6));
+*/
+//#else // IPv4 support
+    struct sockaddr_in clientaddr;
+    clientaddr.sin_family = AF_INET;
+    inet_pton(AF_INET, daemon_local->UDPMulticastIPAddress, &clientaddr.sin_addr);
+    clientaddr.sin_port = htons(daemon_local->UDPMulticastIPPort);
+    memcpy(&g_udpmulticast_addr.clientaddr, &clientaddr, sizeof(struct sockaddr_in));
+//#endif
+    g_udpmulticast_addr.clientaddr_size = sizeof(g_udpmulticast_addr.clientaddr);
+    g_udpmulticast_addr.isvalidflag = ADDRESS_VALID;
+}
+
+// **************************************************************************
+// Function   : dlt_daemon_udp_connection_setup
+// In Param   : contains daemon param values used globally
+// Out Param  : status of udp connection setup and fd registration
+// Description: DataGram socket fd connection is setup
+//                fd is registered with the epoll
+// **************************************************************************
+DltReturnValue dlt_daemon_udp_connection_setup(DltDaemonLocal* daemon_local)
+{
+    int fd = -1;
+    DltReturnValue ret_val = DLT_RETURN_ERROR;
+
+    if(daemon_local == NULL)
+    {
+        return ret_val;
```

Comment:
Should return DLT_RETURN_WRONG_PARAMETER

Reply:
Done